src/main/java/io/github/cottonmc/cotton/gui/widget/WLabel.java

```diff
@@ -29,9 +38,44 @@ public WLabel(String text) {
 	}
 
 	@Override
-	public void paintBackground(int x, int y) {
+	public void paintBackground(int x, int y, int mouseX, int mouseY) {
 		String translated = text.asFormattedString();
 		ScreenDrawing.drawString(translated, alignment, x, y, this.getWidth(), LibGuiClient.config.darkMode ? darkmodeColor : color);
+
+		Text hoveredText = getTextAt(mouseX, mouseY);
+		if (hoveredText != null) {
+			Screen screen = MinecraftClient.getInstance().currentScreen;
+			if (screen instanceof TextHoverRendererScreen) {
+				((TextHoverRendererScreen) screen).renderTextHover(hoveredText, x + mouseX, y + mouseY);
+			}
+		}
+	}
+
+	@Override
+	public void onClick(int x, int y, int button) {
+		Text hoveredText = getTextAt(x, y);
+		if (hoveredText != null) {
+			Screen screen = MinecraftClient.getInstance().currentScreen;
+			if (screen != null) {
+				screen.handleComponentClicked(hoveredText);
+			}
+		}
+	}
+
+	@Environment(EnvType.CLIENT)
+	@Nullable
+	private Text getTextAt(int x, int y) {
+		if (isWithinBounds(x, y)) {
+			int i = 0;
+			for (Text component : text) {
+				TextRenderer renderer = MinecraftClient.getInstance().textRenderer;
+				i += renderer.getStringWidth(component.asFormattedString());
```

Comment:
Are we sure that the sum of the advance widths of each component is in fact the advance width of their combined presentation?
Basically, if we're sure that icu is never called for the full string, only its parts, and that the space (if any) between components is accounted for, then this is appropriate.